eks/mng/scale/scale.go

```diff
@@ -0,0 +1,239 @@
+		)
+	case <-ts.cfg.Stopc:
+		ts.cfg.Logger.Warn("MNG scaler wait aborted; exiting", zap.String("mng-name", mngName))
+		return errors.New("MNG scaler wait aborted")
+	}
+
+	// ref. https://docs.aws.amazon.com/cli/latest/reference/eks/update-nodegroup-config.html
+	var scaleErr = ts.scaleMNG(mngName, update)
+	if scaleErr != nil {
+		return scaleErr
+	}
+
+	ts.cfg.Logger.Info("completed MNG scaler",
+		zap.String("from", ts.cfg.EKSConfig.Parameters.Version),
+		zap.String("to", ts.cfg.EKSConfig.AddOnClusterVersionUpgrade.Version),
+	)
+	return nil
+}
+
+// Scales MNG to the next scaling configuration in the array.
+func (ts *tester) scaleMNG(mngName string, update *eksconfig.MNGScaleUpdate) (err error) {
+	cur := ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] 
+	nodegroupConfigInput := &aws_eks.UpdateNodegroupConfigInput{
+		ClusterName:   aws.String(ts.cfg.EKSConfig.Name),
+		NodegroupName: aws.String(mngName),
+		ScalingConfig: &aws_eks.NodegroupScalingConfig{DesiredSize: aws.Int64(update.DesiredSize),
+			MaxSize: aws.Int64(update.MaxSize),
+			MinSize: aws.Int64(update.MinSize)}}
+	var updateOut *eks.UpdateNodegroupConfigOutput
+	updateOut, err = ts.cfg.EKSAPI.UpdateNodegroupConfig(nodegroupConfigInput)
+
+	if err != nil {
+		ts.cfg.Logger.Warn("MNG scaler request failed", zap.String("mng-name", mngName), zap.Error(err))
+		return err
+	}
+	if updateOut.Update != nil {
+		reqID = aws.StringValue(updateOut.Update.Id)
+	}
+
+	// takes TODO
+	initialWait := 3 * time.Minute
+	totalWait := 5*time.Minute + 20*time.Second*time.Duration(cur.ASGDesiredCapacity)
+
+	ts.cfg.Logger.Info("sent MNG scaler request; polling",
+		zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+		zap.String("mng-name", mngName),
+		zap.String("request-id", reqID),
+		zap.Duration("total-wait", totalWait),
+	)
+
+	// enough time for upgrade fail/rollback
+	ctx, cancel := context.WithTimeout(context.Background(), totalWait)
+	updateCh := wait.PollUpdate(
+		ctx,
+		ts.cfg.Stopc,
+		ts.cfg.Logger,
+		ts.cfg.EKSAPI,
+		ts.cfg.EKSConfig.Name,
+		mngName,
+		reqID,
+		eks.UpdateStatusSuccessful,
+		initialWait,
+		30*time.Second,
+	)
+	for v := range updateCh {
+		err = v.Error
+	}
+	cancel()
+	if err != nil {
+		ts.cfg.Logger.Warn("MNG scale failed when polling", zap.String("mng-name", mngName), zap.Error(err))
+		return err
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), totalWait)
+	nodesCh := wait.Poll(
+		ctx,
+		ts.cfg.Stopc,
+		ts.cfg.Logger,
+		ts.cfg.EKSAPI,
+		ts.cfg.EKSConfig.Name,
+		mngName,
+		aws_eks.NodegroupStatusActive,
+		initialWait,
+		20*time.Second,
+	)
+	for sv := range nodesCh {
+		status := ""
+		if sv.NodeGroup != nil {
+			status = aws.StringValue(sv.NodeGroup.Status)
+		}
+		if sv.Error != nil {
+			cur.Status = fmt.Sprintf("%q failed update with error %v", sv.NodeGroupName, sv.Error)
+		} else {
+			cur.Status = status
+		}
+		if sv.Error != nil && status == aws_eks.NodegroupStatusCreateFailed {
```

Comment:
`wait.Poll` already handles `NodegroupStatusCreateFailed`? See how we poll updates in `eks/mng/version-upgrade`